design/proposal-updateable-specs.md

```diff
@@ -0,0 +1,90 @@
+# Provide a facility to automatically obtain updated specs
+
+## Background
+
+Troubleshoot is limited in that when folks write an update or a new spec for data collection and analysis, that upgrade is not available when Troubleshoot is invoked from within the kots application manager, or from specs stored in a file.  Updating the local file is simple enough, but updating the specs supplied by kots requires an upgrade to kots in order to collect updates to the default spec supplied by kots, and/or an update to the application which is deployed if the spec supplied by the application is changed.  This update requirement delays folks from accessing updated Troubleshoot specs and therefore discourages people from writing new/updated specs.
+
+It is also difficult to justify the tight coupling of Troubleshoot configuration, which is intended to be an entirely client-side concept, and the application manager which is installed to a Kubernetes environment.
+
+## Goals
+
+* Provide the ability to automatically obtain a spec from a source that receives updates, whenever Troubleshoot is run
+* Decouple the provision of default specs from the kots application manager, so that kots upgrades are not required in order to get access to updated specs
+
+## Non Goals
+
+* Maintain the ability for Troubleshoot to run in an airgapped environment
+* maintain compatilbility with existing Troubleshoot specs
+
+## High-Level Design
+
+Add a new field to the Troubleshoot spec definition, which includes a URI used to locate the current spec online.
```

Comment:
change to "includes a URI used to locate additional specs" rather than specify online, and rather than hinting that we replace the existing spec.